core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/multitenancy/MultitenantMessageReceiver.java

```diff
@@ -99,61 +99,78 @@ private void processResponse(MessageContext mainInMsgContext) throws AxisFault {
-                }
+                String tenantDomain;
                 if (tenantRequestMsgCtx.getProperty(MultitenantConstants.TENANT_DOMAIN) != null) {
-                    String tenant = (String)tenantRequestMsgCtx.getProperty(MultitenantConstants.TENANT_DOMAIN);
-                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenant, true);
-                    tenantResponseMsgCtx.setProperty(MultitenantConstants.TENANT_DOMAIN, tenant);
-                }else{
-                    log.warn("Tenant domain is not available in tenant request message context, hence it might not be " +
-                            "set in the thread local carbon context");
-                }
-                tenantResponseMsgCtx.setProperty(MessageContext.TRANSPORT_IN, tenantRequestMsgCtx
-                        .getProperty(MessageContext.TRANSPORT_IN));
-                tenantResponseMsgCtx.setTransportIn(tenantRequestMsgCtx.getTransportIn());
-                tenantResponseMsgCtx.setTransportOut(tenantRequestMsgCtx.getTransportOut());
-
-                tenantResponseMsgCtx.setProperty(MessageContext.TRANSPORT_HEADERS,
-                        mainInMsgContext.getProperty(MessageContext.TRANSPORT_HEADERS));
-
-                tenantResponseMsgCtx.setAxisMessage(
-                        tenantRequestMsgCtx.getOperationContext().getAxisOperation().
-                        getMessage(WSDLConstants.MESSAGE_LABEL_IN_VALUE));
-
-                tenantResponseMsgCtx.setOperationContext(tenantRequestMsgCtx.getOperationContext());
-                tenantResponseMsgCtx.setConfigurationContext(
-                        tenantRequestMsgCtx.getConfigurationContext());
-
-                tenantResponseMsgCtx.setTo(null);
-                tenantResponseMsgCtx.setSoapAction(mainInMsgContext.getSoapAction());
-
-                tenantResponseMsgCtx.setEnvelope(mainInMsgContext.getEnvelope());
-                
-                if(mainInMsgContext.getProperty(MultitenantConstants.PASS_THROUGH_PIPE) != null){
-                	tenantResponseMsgCtx.setProperty(MultitenantConstants.PASS_THROUGH_PIPE, mainInMsgContext.getProperty(MultitenantConstants.PASS_THROUGH_PIPE));
-                	tenantResponseMsgCtx.setProperty(MultitenantConstants.PASS_THROUGH_SOURCE_CONFIGURATION, mainInMsgContext.getProperty(MultitenantConstants.PASS_THROUGH_SOURCE_CONFIGURATION));
-                	tenantResponseMsgCtx.setProperty("READY2ROCK", mainInMsgContext.getProperty("READY2ROCK"));
-                	tenantResponseMsgCtx.setProperty(MultitenantConstants.PASS_THROUGH_SOURCE_CONNECTION, mainInMsgContext.getProperty(MultitenantConstants.PASS_THROUGH_SOURCE_CONNECTION));
+                    tenantDomain =
+                                   (String) tenantRequestMsgCtx.getProperty(MultitenantConstants.TENANT_DOMAIN);
+                } else {
+                    tenantDomain =
+                                   PrivilegedCarbonContext.getThreadLocalCarbonContext()
```

Comment:
Can you check and fix line breaks in lines 104 and 107. Think we can move the previous code line up.